### PR TITLE
Update glossary to properly document greedy string matching

### PR DESCRIPTION
**Pull request by ybot1122**
#2933 is the larger goal

#3359 raises my case for why this specific doc update should be prioritized

### CODE REVIEW

File: docs/Glossary.md [hunk 1]
```diff
@@ -230,7 +230,8 @@ A *route pattern* (or "path") is a string that describes a portion of a URL. Pat
 
   - `:paramName` – matches a URL segment up to the next `/`, `?`, or `#`. The matched string is called a [param](#params)
   - `()` – Wraps a portion of the URL that is optional
-  - `*` – Matches all characters (non-greedy) up to the next character in the pattern, or to the end of the URL if there is none, and creates a `splat` [param](#params)
+  - `*` – Matches all characters (non-greedy) up to the next character in the pattern, or to the end of the URL if there is none, and creates a `splat` [param](/docs/Glossary.md#params)
```
Comment (timdorr) [inline docs/Glossary.md]: Nit: Can you make those links not absolute? That means less work if we reorganize docs in the future, and when building them into a documentation website.

Comment (ybot1122) [inline docs/Glossary.md]: I'll do that tonight. For now, I plan to update just the pattern-matching specific text.

I'll be happy to raise a separate PR to fix all other doc links.

File: docs/guides/RouteMatching.md [hunk 2]
```diff
@@ -10,12 +10,12 @@ A [route](/docs/Glossary.md#route) has three attributes that determine whether o
 React Router uses the concept of nested routes to let you declare nested sets of views that should be rendered when a given URL is invoked. Nested routes are arranged in a tree-like structure. To find a match, React Router traverses the [route config](/docs/Glossary.md#routeconfig) depth-first searching for a route that matches the URL.
 
 ### Path Syntax
-A route path is [a string pattern](/docs/Glossary.md#routepattern) that is used to match a URL (or a portion of one). Route paths are interpreted literally, except for the following special symbols:
+A route path is [a string pattern](../Glossary.md#routepattern) that is used to match a URL (or a portion of one). Route paths are interpreted literally, except for the following special symbols:
```
Comment (timdorr) [inline docs/guides/RouteMatching.md]: Let's not change the other docs right now. I simply meant we should keep the same style as the rest of Glossary.md.

**Comment by ybot1122** [comment]
Ah I should have paid more attention to the rest of the glossary links. I understand now.

**Comment by timdorr** [comment]
Thanks!

**Pull request merged**